### ISSUE TITLE
feature: `IntDecMode` `float64` and `json.Number` modes (`encoding/json` compatibility) 

### ISSUE THREAD

**Issue by extemporalgenome** [fxamacker/cbor#439]
**Is your feature request related to a problem? Please describe.**

`encoding/json`'s behavior when decoding arbitrary numbers to `any` is either `float64`, or `json.Number`, depending on decoder configuration.

`fxamacker/cbor` already has pretty good method-signature compatibility with `encoding/json`, but in the case of untyped decoding, requires additional complexity on the calling code in the form of additional types which must be handled: `int64`, `uint64`, and `*big.Int`.

**Describe the solution you'd like**

- Add an `IntDecMode` value to cause all CBOR integers (including `*big.Int`) to be decoded as `float64` when the destination is `*any`. This would, of course, be lossy for values far from zero.
- Add an `IntDecMode` value to cause all CBOR integers (including `*big.Int`) to be decoded as `json.Number`. This would not be especially efficient, but would provide excellent compatibility with `encoding/json` for existing applications needing arbitrary-precision number support. Internally, this could just use behave like `IntDecConvertNone` with a post-process call to `strconv.FormatInt`, `strconv.FormatUint`, or `(*big.Int).String`, depending on the underlying type; or some specialized binary-to-decimal converter.
- Optional: `JSONCompatDecOptions` and `JSONNumberDecOptions` (or similar) functions, which set these options, as well as setting `DefaultMapType` to `map[string]any`.

**Comment by theory** [comment]
For those looking for solutions, [this comment](https://github.com/fxamacker/cbor/issues/441#issuecomment-2813294047) shows how to preserve json.Number values using the recently-assigned CBOR tag `284`, the [JSON Number String Tag](https://gist.github.com/theory/ef667af1c725240e6e30d525786d58e6).

**Comment by elv-gilles** [comment]
Even when activating support for JSON Number with tag `284`, decoding into a struct that has a numeric field, e.g. int64, currently fails `cannot unmarshal UTF-8 text string into Go struct field cbor_test.TimeSerial.update_time_epoch_sec of type int64`

This would probably need to be handled in `decoder.parseToValue`.

**Comment by benluddy** [comment]
I coincidentally implemented a FloatToInt option yesterday (without a lenient mode) in https://github.com/fxamacker/cbor/pull/777 and saw your backlink this morning (https://github.com/fxamacker/cbor/issues/439#event-6731449597). Is there any interest in spinning off the FloatToInt decode option from the proposed IntDecMode changes and json.Number support (@fxamacker @elv-gilles)?

It's not uncommon for programs in the Kubernetes ecosystem to using `encoding/json` to unmarshal a JSON representation of an API object from a file into `interface{}`, then serialize it into an HTTP request body. Since the stdlib unmarshals even integral-valued JSON numbers to `float64`, and CBOR preserves the distinction between integers and floating point numbers on the wire, the API server needs to be able to accept integral-valued CBOR floats when decoding them into int/uint fields.

**Comment by elv-gilles** [comment]
@benluddy - In my opinion, these changes go together but I feel a bit conflicted: 
* on one hand, a FloatToInt option would already be a good first step (with lenient mode to help preventing random failures)
* on the other hand, configuring the API server with `useNumber` would look more like the real fix

**Comment by fxamacker** [comment]
> Is there any interest in spinning off the FloatToInt decode option from the proposed IntDecMode changes and json.Number support (@fxamacker @elv-gilles)?

@benluddy Yes, I think FloatToInt would be useful to improve JSON and CBOR interoperability.

@elv-gilles Apologies for the late reply.  I took a quick look at your branch and will need to take a closer look.
